Fix errors in 3->4 migration

OldStarchy opened the pull request:
See comment and issue https://github.com/jonom/silverstripe-focuspoint/issues/89#issuecomment-736099352.

I tried copying values from `Image` to `Image_Live` and `Image_Versions` when these tables are new but the `SQLUpdate` doesn't support cross-table updates (though I've left the code in as comments so you can see what I tried).

I've also moved the `ALTAR TABLE` statements out of `withTransaction` since [they prematurely `COMMIT;` the transaction](https://stackoverflow.com/a/22806756/3094008) causing an error when leaving the `withTransaction` block.

jonom commented on src/Dev/FocusPointMigrationTask.php:
```diff
@@ -1,7 +1,8 @@
 <?php
 
-namespace JonoM\FocusPoint\Dev;
+declare(strict_types=1);
```
Is this necessary? I'm wondering if it could have any unintended side effects.

jonom commented on src/Dev/FocusPointMigrationTask.php:
```diff
@@ -49,36 +50,72 @@ protected function changeDbFields($from, $to, $message)
             $imageTable,
-            $imageTable . "_" . Versioned::LIVE,
-            $imageTable . "_Versions",
+            $imageTable . '_' . Versioned::LIVE,
```
Are the quote changes a PSR thing or just personal preference?

OldStarchy replied:
another php-cs-fixer change

jonom commented on src/Dev/FocusPointMigrationTask.php:
```diff
@@ -49,36 +50,72 @@ protected function changeDbFields($from, $to, $message)
 
-        DB::get_conn()->withTransaction(function() use ($imageTables, $from, $to, $message) {
+        $success = false;
+        DB::get_conn()->withTransaction(function () use ($imageTables, /* $imageTable, */ $from, $to, &$success) {
```
Comment to clean up here

jonom commented on src/Dev/FocusPointMigrationTask.php:
```diff
@@ -49,36 +50,72 @@ protected function changeDbFields($from, $to, $message)
+                //     $query->execute();
+
+                //     DB::get_schema()->alterationMessage("Joined {$table} to {$imageTable} to update {$newColumnX} and {$newColumnY}", 'changed');
+                // } else {
```
Can you remove the commented part please and then should be good to go.

jonom reviewed (changes requested):
Awesome! I did a fresh install of SS3 with focuspoint and uploaded an image, didn't get the error when I upgraded to SS4 and the focus point was preserved. So it seems to do the trick. Just a little clean up to do and then we can merge it in. Thanks so much!

OldStarchy commented:
I was editing this code in another project so my php-cs-fixer config was active and changed a few other things.

I've removed the `strict_types` and the commented code.
`"$foo"` and `"${foo}"` are deprecated in favor of `"{$foo}"` in 8.2.
There was a mix of single and double quotes in use, so I've left the php-cs-fixer change to convert them all to single quotes where possible.

jonom commented:
Thanks! Tagged 4.0.5
